WordPress/Classes/ViewRelated/Menus/Views/MenuItemAbstractPostsViewController.m

```diff
@@ -0,0 +1,201 @@
+        [self performResultsControllerFetchRequest];
+        [self.tableView reloadData];
+    }
+    [self hideLoadingSourcesIndicator];
+}
+
+#pragma mark - paging
+
+- (void)scrollingWillDisplayEndOfTableView:(UITableView *)tableView
+{
+    if (!self.additionalPostsAvailableForSync || [self searchBarInputIsActive]) {
+        return;
+    }
+    [self showLoadingSourcesIndicator];
+    PostService *service = [[PostService alloc] initWithManagedObjectContext:[self managedObjectContext]];
+    PostServiceSyncOptions *options = [self syncOptions];
+    options.offset = @(self.resultsController.fetchedObjects.count);
+    [service syncPostsOfType:[self sourceItemType]
+                 withOptions:options
+                     forBlog:[self blog]
+                     success:^(NSArray *posts) {
+                         [self didFinishSyncingPosts:posts options:options];
+                     } failure:^(NSError *error) {
+                         [self didFinishSyncingPosts:nil options:options];
+                         [self showLoadingErrorMessageForResults];
+                     }];
+}
+
+#pragma mark - searching
+
+- (void)searchBarInputChangeDetectedForLocalResultsUpdateWithText:(NSString *)searchText
+{
+    NSPredicate *defaultPredicate = [self defaultFetchRequestPredicate];
+    NSPredicate *predicate = nil;
+    
+    if (searchText.length) {
+        NSPredicate *searchPredicate = [NSPredicate predicateWithFormat:@"postTitle CONTAINS[cd] %@", searchText];
+        predicate = [NSCompoundPredicate andPredicateWithSubpredicates:@[defaultPredicate, searchPredicate]];
+        if ([self.resultsController.fetchRequest.predicate isEqual:predicate]) {
+            // same predicate, no update needed
+            return;
+        }
+        if (self.additionalPostsAvailableForSync) {
+            self.defersFooterViewMessageUpdates = YES;
+        }
+    } else {
+        predicate = defaultPredicate;
+    }
+    
+    DDLogDebug(@"MenuItemSourcePostView: Updating fetch request predicate");
+    self.resultsController.fetchRequest.predicate = predicate;
+    [self performResultsControllerFetchRequest];
+    [self.tableView reloadData];
+}
+
+- (void)searchBarInputChangeDetectedForRemoteResultsUpdateWithText:(NSString *)searchText
+{
+    if (!searchText.length || !self.additionalPostsAvailableForSync) {
+        self.defersFooterViewMessageUpdates = NO;
+        return;
+    }
+    
+    self.defersFooterViewMessageUpdates = NO;
+    
+    [self showLoadingSourcesIndicator];
+    void(^stopLoading)() = ^() {
+        [self hideLoadingSourcesIndicator];
+    };
+    
+    DDLogDebug(@"MenuItemSourcePostView: Searching posts via PostService");
+    PostService *service = [[PostService alloc] initWithManagedObjectContext:[self managedObjectContext]];
+    PostServiceSyncOptions *options = [self syncOptions];
+    options.search = searchText;
+    [service syncPostsOfType:[self sourceItemType]
```

Comment:
I *really* wish we had a good way to cancel an inflight remote search vs just ignoring it :(